sugar.go

```diff
@@ -74,7 +74,14 @@ func Desugar(s *SugaredLogger) Logger {
 // continues. Passing an odd number of arguments triggers similar behavior:
 // panics in development and errors in production.
 func (s *SugaredLogger) With(args ...interface{}) *SugaredLogger {
-	return &SugaredLogger{core: s.core.With(s.sweetenFields(args)...)}
+	return s.WithFields(s.sweetenFields(args)...)
+}
+
+// WithFields adds structured fields to the logging context, much like the
+// base Logger. It allows the sugared logger to use more specialized
+// fields (like Stack).
+func (s *SugaredLogger) WithFields(fs ...zapcore.Field) *SugaredLogger {
```

Comment:
Did you mean to just slip this one into this PR? It seems like a whole different thing; also it's attached to a larger ball of wax:
- if we're going to cede that it's useful toe have a `FooFields` for `Foo=With`...
- ...why not also for `Foo` in `{Debug,Info,Warn,Error,DPanic,Panic,Fatal}`?
- At that point, why don't we just collapse `SugaredLogger` into `Logger`, and dispense with all the wrapping?
- The user story would then become something like: "most people use Foof / Foow, perf-conscious users use FooFields"; maybe even something like:
  - "Don't care about logging? Use Foow"
  - "Want structured logs? Use Foof"
  - "Really care about logs? Library author? Use FooFields"